tests/test_point.py

```diff
@@ -139,7 +142,9 @@ def test_coords(self):
 def test_empty_point_bounds():
     """The bounds of an empty point is an empty tuple"""
     p = Point()
-    assert p.bounds == ()
+    # TODO keep this change or not?
+    assert len(p.bounds) == 4
+    assert all(math.isnan(v) for v in p.bounds)
```

Comment:
Why does just `point empty` have four nan bounds, and not other empty geometry types (that return an empty tuple `()`)? Otherwise, yes this idea is worth considering.

Reply:
Ah, that's an oversight / something to clean-up. For the base class method, I left the current behaviour but added a TODO comment:

```diff
     @property
     def bounds(self):
         """Returns minimum bounding region (minx, miny, maxx, maxy)"""
+        # TODO return empty tuple or (nan, nan, nan, nan)
         if self.is_empty:
             return ()
         else:
-            return self.impl['bounds'](self)
+            return tuple(pygeos.bounds(self).tolist())
```

while this property is overridden by Point, where I actually removed this empty tuple return:

```diff
    @property
    def bounds(self):
        """Returns minimum bounding region (minx, miny, maxx, maxy)"""
-        try:
-            xy = self.coords[0]
-        except IndexError:
-            return ()
-        return (xy[0], xy[1], xy[0], xy[1])
+        # TODO keep return type of tuple, or just numpy array?
+        return tuple(pygeos.bounds(self).tolist())
```

So for sure, we need to be consistent about this, whatever decision we make. 

(and actually, with the new implementation, we can remove the override in the Point subclass, as the base class implementation should work fine for Points as well)

Reply:
Removed the override in `Point` class, and leaving the discussion to changy `empty.bounds` for a separate issue

Reply:
Opened https://github.com/Toblerity/Shapely/issues/1023 to discuss this